lib/next_ls/db.ex

```diff
@@ -13,8 +13,8 @@ defmodule NextLS.DB do
   @spec query(pid(), query(), list()) :: list()
   def query(server, query, args \\ []), do: GenServer.call(server, {:query, query, args}, :infinity)
 
-  @spec symbols(pid()) :: list(map())
-  def symbols(server), do: GenServer.call(server, :symbols, :infinity)
+  @spec symbols(pid(), [String.t()]) :: list(map())
+  def symbols(server, workspace_folders), do: GenServer.call(server, {:symbols, workspace_folders}, :infinity)
```

Comment:
this function is actually a residual from the pre-sqlite days, and I think we can "unpack" it now. (passing in the workspace folders to me is leaking implementation details and is a sign we want to yield full control back to the caller)

so i would delete this function and associated callback, and move the query it calls to the callsite, and use the query function instead.